python/ray/serve/metric/exporter.py

```diff
@@ -0,0 +1,144 @@
+from typing import Dict
+from collections import Counter, namedtuple
+
+import ray
+from ray.serve.metric.types import MetricType, MetricMetadata, MetricBatch
+from ray.serve.utils import _get_logger
+
+logger = _get_logger()
+
+
+def make_metric_namedtuple(metric_metadata: MetricMetadata,
+                           record: MetricBatch):
+    fields = ["name", "type"]
+    fields += list(metric_metadata.default_labels.keys())
+    fields += list(record.labels.keys())
+
+    merged_labels = metric_metadata.default_labels.copy()
+    merged_labels.update(record.labels)
+
+    tuple_type = namedtuple(metric_metadata.name, fields)
+    return tuple_type(
+        name=metric_metadata.name, type=metric_metadata.type, **merged_labels)
+
+
+class BaseExporter:
+    """Aggregate metrics from all RayServe actors and export them.
+
+    Metrics are aggregated pushed from other actors in the system to
+    this actor. They can then be stored or pushed to an external monitoring
+    system.
+    """
+
+    def __init__(self):
+        # Stores the mapping metric_name -> MetricMetadata
+        # This field is tolerant to failures since each client will always push
```

Comment:
Will each client push an updated copy of the metadata with every message?

Reply:
Yes